### ISSUE TITLE
Odd formatting with if statements

### ISSUE THREAD

**Issue by carllerche** [rust-lang/rustfmt#3288]
We tried running rustfmt against the http crate. It mostly acted as expected. However, the following change was unexpected to me:

Before:

```rust
} else if eq!(b[1] == b'o' b'n' b't' b'e' b'n' b't' b'-' b'r' b'a' b'n' b'g' b'e' ) {
```

After:

```rust
} else if eq!(b[1] == b'o' b'n' b't' b'e' b'n' b't' b'-' b'r' b'a' b'n' b'g' b'e' )
{
```

[source](https://github.com/hyperium/http/pull/289/files#diff-a26f9382eb4fa7d9d2b7a2b4a076dddeR1288).

I am not sure if this is expected or a bug in rustfmt.

Edit: woops, fixed copy / paste fail. Thanks @scampi.

**Comment by scampi** [comment]
The source is https://github.com/hyperium/http/pull/289/files#diff-a26f9382eb4fa7d9d2b7a2b4a076dddeR1288, you pointed to another issue instead ;o)
The length is over 100 so it tries to wrap somehow. If you used the default [brace style](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#control_brace_style), then maybe that could be an exception and it would go over the limit ? @nrc is that possible ?

**Comment by carllerche** [comment]
@scampi right, I understand the logic that probably got this behavior. What I am wondering is if it is intentional.

This is using rustfmt's default configuration.

The case seems to be a macro that cannot be formatted (probably due to invalid rust?) that is over 100 chars. This means the only thing that can be wrapped is the curly bracket.

My expectation would be that the curly bracket would not be the *only* thing that is wrapped.

**Comment by topecongiro** [comment]
> The case seems to be a macro that cannot be formatted (probably due to invalid rust?)

True.

> What I am wondering is if it is intentional.

This is intentional. Since the macro cannot be formatted by rustfmt, it needs to get kept as is, so the only other option is to put the condition on the next line:

```rust
} else if
    eq!(b[1] == b'o' b'n' b't' b'e' b'n' b't' b'-' b'r' b'a' b'n' b'g' b'e' )
{
```
```rust
// or
} else if
    eq!(b[1] == b'o' b'n' b't' b'e' b'n' b't' b'-' b'r' b'a' b'n' b'g' b'e' ) {
```

IMHO putting the opening brace on the next line is better than above.

**Comment by carllerche** [comment]
@topecongiro neither examples are what rustfmt do today.

**Comment by topecongiro** [comment]
@carllerche Sorry if I confused you, I mean that the current behavior (putting the opening brace on the next line if the `if` line barely exceeds max width) is better than the examples in my last post.

**Comment by ytmimi** [comment]
@calebcartwright, I'm not sure if there's any action to be taken here. I feel like we should close this.

**Comment by ytmimi** [comment]
smallest reproduction I could come up with is as follows:

**Input**:

```rust
fn main() {
    match x {
        _ => {
            if true {
            } else if true {
                if true {
                } else if eq!(b[1] == b'o' b'n' b't' b'e' b'n' b't' b'-' b'r' b'a' b'n' b'g' b'e' ) {
                    return Ok(ContentRange.into())
                }
            }
        }
    }
}
```

**Output**

```rust
fn main() {
    match x {
        _ => {
            if true {
            } else if true {
                if true {
                } else if eq!(b[1] == b'o' b'n' b't' b'e' b'n' b't' b'-' b'r' b'a' b'n' b'g' b'e' )
                {
                    return Ok(ContentRange.into());
                }
            }
        }
    }
}
```

This has been open for a while, and I don't think there's any action for the team to take. The `{` gets wrapped to keep this under the `max_width`.
